src/gen_bpf.c

```diff
@@ -569,11 +572,16 @@ static int _hsh_add(struct bpf_state *state, struct bpf_blk **blk_p,
 		return -ENOMEM;
 
 	/* generate the hash */
-	h_val = hash(blk->blks, _BLK_MSZE(blk));
+	h_val_tmp[0] = hash(blk->blks, _BLK_MSZE(blk));
+	h_val_tmp[1] = hash(&blk->acc_start, sizeof(blk->acc_start));
+	h_val_tmp[2] = hash(&blk->acc_end, sizeof(blk->acc_end));
+	h_val = hash(h_val_tmp, sizeof(h_val_tmp));
 	blk->hash = h_val;
 	blk->flag_hash = true;
 	blk->node = NULL;
 	h_new->blk = blk;
+	h_new->acc_start = blk->acc_start;
+	h_new->acc_end = blk->acc_end;
```

Comment:
I actually liked what you were doing in your patch where you simply reused the accumulator state kept inside the `blk` variable.  I'd vote for changing this to use that approach, or was there a problem with that?

Reply:
I don't believe there were any issues with the approach I had taken, but I admit I would like to get more testing miles on it.

I had forgotten to push your latest patch to my git repo and was simply running with my intermediate one.  I'll grab the reused accumulator changes from my patch and merge them with this one.  I'll try to log some more testing miles as well.

Stay tuned...

Reply:
Great, thanks!